src/d3d12/d3d12_resource_pool_texture.cpp

```diff
@@ -542,6 +542,235 @@ namespace wr
 		return texture;
 	}
 
+	d3d12::TextureResource * D3D12TexturePool::LoadPNGFromMemory(char * data, size_t size, bool srgb, bool generate_mips)
```

Comment:
I didn't see this in the first review, but the load functions have been modified recently, there's some pieces of code that you are missing from this one. An example is the code that adjustes the format that you can see in the LoadPNG function, also the set name call on the texture resouce. If you don't have time to do this, write it in a comment to this one and I'll do it as soon as I have time